### PR TITLE
Migrate cometBorrow and cometRepay to use Dynamic list [p2]

### PR DESCRIPTION
**Pull request by cwang25**
- Prerequisite for adding auto-wrapper features to those two functions.

### CODE REVIEW

File: src/builder/QuarkBuilder.sol [hunk 1]
```diff
@@ -152,41 +151,43 @@ contract QuarkBuilder {
...
-                QuarkOperationHelper.wrapOperationsWithTokenPayment(quarkOperations, actions, payment, useQuotecall);
+            quarkOperationsArray = QuarkOperationHelper.wrapOperationsWithTokenPayment(
+                quarkOperationsArray, actionsArray, payment, useQuotecall
+            );
         }
 
         // Construct EIP712 digests
         EIP712Helper.EIP712Data memory eip712Data;
-        if (quarkOperations.length == 1) {
+        if (quarkOperationsArray.length == 1) {
```
Comment (scott-silver) [inline src/builder/QuarkBuilder.sol:171]: This whole code block was deleted from `main` in #51. I'm guessing that if you merge p2 into p1 that you're going to have merge conflicts.

But if you rebase, you should be able to just delete any changes you've made here to lines 170-183.

File: src/builder/QuarkBuilder.sol [hunk 2]
```diff
@@ -326,41 +325,43 @@ contract QuarkBuilder {
...
```
Comment (scott-silver) [inline src/builder/QuarkBuilder.sol:345]: same comment as above; lines 343-359 no longer exist on main. You should be able to rebase and then remove those changes

File: src/builder/QuarkBuilder.sol (hunk 1 above)
Comment (cwang25) [reply src/builder/QuarkBuilder.sol]: Oh I see.  Thanks! I have rebased it so this change is now gone 👍

File: src/builder/QuarkBuilder.sol (hunk 2 above)
Comment (cwang25) [reply src/builder/QuarkBuilder.sol]: Oh I see.  Thanks! I have rebased it so this change is now gone 👍

**Review by kevincheng96 — APPROVED** [review]

**Pull request merged**